datasets/noaa/pipelines/_images/run_csv_transform_kub/csv_transform.py

```diff
@@ -0,0 +1,1043 @@
+    int_date_list: typing.List[str],
+    gen_location_list: dict,
+) -> None:
+    if pipeline_name == "GHCND by year":
+        if full_data_load == "N":
+            start = str(datetime.datetime.now().year - 6)
+        else:
+            start = start_year
+        ftp_batch = 1
+        for yr in range(int(start), datetime.datetime.now().year + 1):
+            yr_str = str(yr)
+            source_zipfile = str.replace(str(source_file), ".csv", f"_{yr_str}.csv.gz")
+            source_file_unzipped = str.replace(str(source_zipfile), ".csv.gz", ".csv")
+            target_file_year = str.replace(str(target_file), ".csv", f"_{yr_str}.csv")
+            destination_table_year = f"{destination_table}_{yr_str}"
+            source_url_year = str.replace(source_url, ".csv.gz", f"{yr_str}.csv.gz")
+            target_gcs_path_year = str.replace(
+                target_gcs_path, ".csv", f"_{yr_str}.csv"
+            )
+            if ftp_batch == int(ftp_batch_size):
+                logging.info("Sleeping...")
+                time.sleep(int(ftp_batch_sleep_time))
+                ftp_batch = 1
+            else:
+                ftp_batch += 1
+            download_file_ftp(
+                ftp_host=ftp_host,
+                ftp_dir=ftp_dir,
+                ftp_filename=f"{yr_str}.csv.gz",
+                local_file=source_zipfile,
+                source_url=source_url_year,
+            )
+            gz_decompress(
+                infile=source_zipfile, tofile=source_file_unzipped, delete_zipfile=True
+            )
+            process_and_load_table(
+                source_file=source_file_unzipped,
+                target_file=target_file_year,
+                pipeline_name=pipeline_name,
+                source_url=source_url_year,
+                chunksize=chunksize,
+                project_id=project_id,
+                dataset_id=dataset_id,
+                destination_table=destination_table_year,
+                target_gcs_bucket=target_gcs_bucket,
+                target_gcs_path=target_gcs_path_year,
+                schema_path=schema_path,
+                drop_dest_table=drop_dest_table,
+                input_field_delimiter=input_field_delimiter,
+                input_csv_headers=input_csv_headers,
+                data_dtypes=data_dtypes,
+                reorder_headers_list=reorder_headers_list,
+                null_rows_list=null_rows_list,
+                date_format_list=date_format_list,
+                slice_column_list=slice_column_list,
+                regex_list=regex_list,
+                rename_headers_list=rename_headers_list,
+                remove_source_file=remove_source_file,
+                delete_target_file=delete_target_file,
+                int_date_list=int_date_list,
+                gen_location_list=gen_location_list,
+            )
+    if pipeline_name in [
+        "GHCND countries",
+        "GHCND inventory",
+        "GHCND states",
+        "GHCND stations",
```

Comment:
I think it is better to concert this to a list (or a dictionary) of constants and move it at the top of this file (or put it in a yaml file and read from it).

Reply:
What added value would that bring to the code?  What would the variable be called?